broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -258,28 +256,31 @@ public void onUnrecoverableFailure() {
     actor.run(this::handleUnrecoverableFailure);
   }
 
-  private void onInstallFailure(final long term, final Throwable error) {
+  private void onInstallFailure(final Throwable error) {
     if (error instanceof UnrecoverableException) {
       LOG.error(
           "Failed to install partition {} with unrecoverable failure: ",
           context.getPartitionId(),
           error);
       handleUnrecoverableFailure();
     } else {
-      handleRecoverableFailure(term);
+      handleRecoverableFailure();
     }
   }
 
-  private void handleRecoverableFailure(final long term) {
+  private void handleRecoverableFailure() {
     zeebePartitionHealth.setServicesInstalled(false);
     context
         .getPartitionListeners()
-        .forEach(l -> l.onBecomingInactive(context.getPartitionId(), term));
+        .forEach(l -> l.onBecomingInactive(context.getPartitionId(), context.getCurrentTerm()));
 
-    if (context.getRaftPartition().getRole() == Role.LEADER) {
+    // If RaftPartition has already transition to a new role in a new term, we can ignore this
+    // failure. Services will be installed for the new role.
+    if (context.getCurrentRole() == Role.LEADER
+        && context.getCurrentTerm() == context.getRaftPartition().term()) {
```

Comment:
Lets say the node become leader in raft. The previous follower installation failed, for what ever reason. Then the term is not equal and the currentRole is Follower, we going to inactive, instead of just installing the leader partition. Does this make sense?

Why we ignore the failure when we become leader in a higher term? This would mean we have a running raft leader but not Zeebe Leader Partition installed?

Reply:
> Lets say the node become leader in raft. The previous follower installation failed, for what ever reason. Then the term is not equal and the currentRole is Follower, we going to inactive, instead of just installing the leader partition. Does this make sense?

If the previous follower installation failed, raft would go inactive.

> Why we ignore the failure when we become leader in a higher term? This would mean we have a running raft leader but not Zeebe Leader Partition installed?

Higher term in raft means there is already a new transition enqueued in ZeebePartition. Hence the services for that transition will be installed. If we don't ignore this failure and stepdown, then we are unnecessarily triggering a new leader election.

Reply:
> If the previous follower installation failed, raft would go inactive.

I get that, but why? We could also just install the leader partition right?

> Higher term in raft means there is already a new transition enqueued in ZeebePartition. Hence the services for that transition will be installed. If we don't ignore this failure and stepdown, then we are unnecessarily triggering a new leader election.

Thanks makes sense now. Maybe we can make the doc a bit more clearer. As you wrote before with higher term, we know that a transition is enqueue which is why it is safe to ignore it etc.

Reply:
> I get that, but why? We could also just install the leader partition right?

Sorry. I misunderstood your question. :facepalm: 
We can ignore the failure if (only if) the new role is leader. However there is a scenario that can lead to no follower services installed in zeebe.
1. Raft transition to leader (term 2). But the initial entry is not committed. Raft doesn't notify zeebe until initial entry is committed.
2. Follower install failed (term 1)
3. Raft initial entry commit failed. Raft transition to follower in term 2.
4. Raft notifies zeebe of follower role.
5. Zeebe doesn't install follower services because it is already in follower role.